spellbook/models/sudoswap/ethereum/sudoswap_ethereum_events.sql

```diff
@@ -0,0 +1,240 @@
+            , call_block_number as block_number
+            , token_id
+            , 'ERC721' as token_standard
+            , cardinality(token_id) as number_of_items
+            , CASE WHEN cardinality(token_id) > 1 THEN 'Bundle Trade'
+                ELSE 'Single Item Trade' 
+               END as trade_type
+            , trade_category
+            , 'Trade' as evt_type
+            , CASE WHEN trade_category = 'Buy' THEN pair_address --AMM is selling if an NFT is being bought
+                ELSE call_from
+                END as seller 
+            , CASE WHEN trade_category = 'Sell' THEN pair_address --AMM is buying if an NFT is being sold
+                ELSE call_from
+                END as buyer
+            , trade_price_eth as amount_raw
+            , trade_price_eth/1e18 as amount_original
+            , '0x0000000000000000000000000000000000000000' as currency_contract --ETH
+            -- amount_usd
+            , nftcontractaddress as nft_contract_address
+            , '0xb16c1342e617a5b6e4b631eb114483fdb289c0a4' as project_contract_address --not sure what this is? I put their main factory for now
+            , call_tx_hash as tx_hash
+            , null as evt_index --we didn't use events in our case for decoding, so this will be null until we find a way to tie it together.
+            , base_price*protocolfee as platform_fee_amount_raw
+            , (base_price*protocolfee)/1e18 as platform_fee_amount
+            , protocolfee as platform_fee_percentage
+            --royalties don't technically exist on AMM, but there are owner fees for the pool that can be routed as royalties in the future.
+            , base_price*ownerfee as royalty_fee_amount_raw
+            , (base_price*ownerfee)/1e18 as royalty_fee_amount
+            , ownerfee as royalty_fee_percentage
+            , pair_address as royalty_fee_receive_address
+            , 'ETH' as royalty_fee_currency_symbol
+        FROM swaps_w_traces
+    ),
+
+    swaps_cleaned_w_metadata as (
+        SELECT 
+            sc.*
+            , tokens.name AS collection
+            , agg.name as aggregator_name
+            , agg.contract_address as aggregator_address
+            , sc.amount_original*pu.price as amount_usd 
+            , sc.royalty_fee_amount*pu.price as royalty_fee_amount_usd
+            , sc.platform_fee_amount*pu.price as platform_fee_amount_usd
+            , tx.from 
+            , tx.to 
+            , 'sudoswap-' || sc.tx_hash || '-' || sc.nft_contract_address || sc.token_id::string || '-' || sc.seller || '-' || sc.amount_original::string || 'Trade' AS unique_trade_id
+        FROM swaps_cleaned sc
+        LEFT JOIN prices_usd_eth pu ON pu.blockchain='ethereum'
+            AND date_trunc('minute', pu.minute)=date_trunc('minute', sc.block_time)
+            --add in `pu.contract_address = sc.currency_address` in the future when ERC20 pairs are added in.
+        LEFT JOIN eth_transactions tx ON tx.hash=sc.tx_hash
```

Comment:
after the CTE cleanup, i'd ensure this is filtered as much as possible. we're looking into more performance improvements related to base table joins on tx_hash